cmd/erasure-object.go

```diff
@@ -1912,23 +1912,26 @@ func (er erasureObjects) DeleteObject(ctx context.Context, bucket, object string
 	versionFound := true
 	objInfo = ObjectInfo{VersionID: opts.VersionID} // version id needed in Delete API response.
 	goi, _, gerr := er.getObjectInfoAndQuorum(ctx, bucket, object, opts)
+	tryDel := false
 	if gerr != nil && goi.Name == "" {
 		if _, ok := gerr.(InsufficientReadQuorum); ok {
-			// Add an MRF heal for next time.
-			er.addPartial(bucket, object, opts.VersionID)
-
-			return objInfo, InsufficientWriteQuorum{}
+			if opts.Versioned || opts.VersionSuspended || countOnlineDisks(storageDisks) < len(storageDisks)/2+1 {
```

Comment:
It would be good to handle this without the `tryDel` boolean. It looks like the current code implicitly assumes that an error other than InsufficientReadQuorum is VersionNotFound, as evidenced by `if opts.DeleteMarker { versionFound = false ...`. If we make the error handling for getObjectInfoAndQuorum more robust, we can avoid this boolean flag to alter control flow. If you feel that this change is out of scope of what you wish to accomplish in this change, that is fine too. We can address that subsequently.